core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -94,6 +98,16 @@ enum Consider {
 	RejectPast,
 	/// Message is from the future. Reject.
 	RejectFuture,
+    /// Message is redundant. Reject.
+    RejectRedundant,
+	RejectEquivocation,
```

Comment:
indentation is still wrong. I don't see the purpose of `RejectEquivocation`. can you explain?

Reply:
> We do want to reject all redundant equivocations, but we only want to punish peers based on the combination of what they've sent to us

`RejectEquivocation` is meant for the "punish peers based on the combination of what they've sent to us" case, whereas `RejectRedundant` is meant to reject a redundant equivocation, without downgrading the peer. 

I guess the naming is off?

Reply:
Yeah, the naming is really off.